### PR TITLE
Change the text wrapping for text display.

### LINKED ISSUE #1951: Edit Mode of Text Box is Pushing letters to new line
## Bug Description

Edit mode and display mode should display the same way -- if the letters are on a new line in display mode, it should also happen for edit mode. If the word is not breaking in display mode, it shouldn't break for edit mode either.

## Steps to Reproduce

1. Size a textbox with text so that the last letter is right up to the edge of the selection box
2. exit edit mode
3. enter edit mode
4. see letters pushed to new line when entering edit mode and text is highlighted

## Screenshots

https://docs.google.com/presentation/d/1ZiZWrmLiX40rt3n_1LMSGJxYWI-xsWFsS9bkzdbRIV8/edit?ts=5ec3ef17#slide=id.g7830109f71_0_48

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

Update the resizing based on [Figma](https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=8011%3A14593) -- add wrapping to overflowing text.

### PR DESCRIPTION
**Pull request by miina**
## Summary

Changes the Text wrapping for a text element.
Per [Figma](https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=8011%3A14593), when Text's width is resized smaller, the words shouldn't overflow.

We had the text just overflowing but only in the display mode which caused differences between the edit mode and the display mode.

## Relevant Technical Choices

- Rounds width and height used for display mode since otherwise there are differences with the measurer which can cause the height not being accurately calculated.

## User-facing changes
- When text element's width is resized, the words will be wrapped when they don't fit in length.

## Testing Instructions
I
1. Add a text element
2. Add a longer word in
3. Resize the width of the element smaller and wider again
4. Verify that the longer word is breaking into multiple lines and not overflowing
5. Enter edit mode, verify that the view is the same for edit mode and display mode -- the word is breaking the same way.

II
1. Do the same steps again but this time use Highlight as the background mode for the Text element.
---

Fixes #1951

### CODE REVIEW

File: assets/src/edit-story/elements/shared/index.js [hunk 1]
```diff
@@ -69,6 +70,7 @@ export const elementWithTextParagraphStyle = css`
   padding: ${({ padding }) => padding || 0};
   line-height: ${({ lineHeight }) => lineHeight};
   text-align: ${({ textAlign }) => textAlign};
+  overflow-wrap: anywhere;
```
Comment (swissspidy) [inline assets/src/edit-story/elements/shared/index.js]: TIL that `word-wrap` was renamed to `overflow-wrap`!

**Review by barklund — COMMENTED** [review]
Generally this looks good.

I can however still create a situation with a different number of lines in display-vs-output (but same in display-vs-edit). Also, the measurer clearly uses a different word-breaking rule then the display here (background box has wrong size):

Screen Shot 2020-07-10 at 16 14 57
_Editor on the left, output on the right_

**Comment by miina** [comment]
> Also, the measurer clearly uses a different word-breaking rule then the display here (background box has the wrong size):

@barklund Actually, the measurer uses [the same word-breaking rule](https://github.com/google/web-stories-wp/pull/3125/files#diff-1382841cabbcfb2cadd4b745d94770a8R36) (measurer in red):
Screenshot 2020-07-13 at 11 21 10


However, there seems to be an issue, potentially with conversion between pixels and data units, which creates this difference.

Screenshot 2020-07-13 at 11 20 59


The measurer is using the Text Output, meaning that the measurer matches what's seen in the front-end, however, apparently, that does not match the data units -- the measurer has a different size in pixels than the actual display as can be seen from the screenshots.


Not sure what's the best approach here -- if the measurer would match the display instead, that would potentially cause differences with the front-end. For example, we could use the scroll-height of the display directly instead of using a separate measurer for this.

This is be a bit out of scope though since the same happens on the main branch:
Screenshot 2020-07-13 at 11 27 39

... And this issue was about unifying the edit and display mode, it might make sense to create a separate issue to figure out what to do with the measurer vs editor. Thoughts?

**Comment by merapi** [comment]
> create a separate issue to figure out what to do with the measurer vs editor. Thoughts?

Today I will push a fix for this issue. [This one](https://github.com/google/web-stories-wp/issues/835) is about the same thing?

**Comment by merapi** [comment]
How about changing `anything` to `break-word`?
Looks like `min-content intrinsic size` is irrelevant for us and we will match the DraftJS rule + it will work in Safari.

https://codepen.io/merapi/full/qBbyVQP
Screen Shot 2020-07-13 at 13 14 21

**Comment by miina** [comment]
> Today I will push a fix for this issue. This one is about the same thing?

Ok, great, thank you! So we can leave that part out of this PR?

Not sure about #835, it doesn't have enough information to under the issue to understand what's the exact issue.

**Comment by miina** [comment]
> How about changing anything to break-word?

Good point, done!

**Review by merapi — APPROVED** [review]

**Review by barklund — APPROVED** [review]

**Pull request merged**